### PR TITLE
Sending mail to people from same institute

### PR DESCRIPTION
**Pull request by gauthamzz**

### CODE REVIEW

File: models/db.py [hunk 1]
```diff
@@ -184,7 +184,37 @@ def sanitize_fields(form):
     for site in current.SITES:
         site_handle = site.lower() + "_handle"
         form.vars[site_handle] = form.vars[site_handle].lower()
-
+#-----------------------------------------------------------------------------
+def inside_institute(form):
+        # Send mail to all users form the same college when someone registers
+    to = []
+    for mail in db(utable.email == form.vars.institute).select():
```
Comment (raj454raj) [inline models/db.py]: Okay this line says you haven't tested the code.
`utable` not defined - Should be atable, and `atable = db.auth_user`
and the query should be 
```
db(atable.institute == form.vars.institute).select(atable.email)
```
The `mail` variable could be renamed as `email_id`
Finally keep this query in a seperate variable(say varname is `rows`)
And then 
```
for email_id in rows:
```
would be a cleaner way.

File: models/db.py [hunk 2]
```diff
@@ -184,7 +184,37 @@ def sanitize_fields(form):
...
+        to.append(mail)
```
Comment (raj454raj) [inline models/db.py]: this would become row.email

File: models/db.py [hunk 3]
```diff
@@ -184,7 +184,37 @@ def sanitize_fields(form):
...
+    subject = "New user registered from your college"
+    message = """
```
Comment (raj454raj) [inline models/db.py]: Okay sorry here - I didn't mention what should the mail content look like.
I want something like this to be seen in the mails - 
```
Hello <1>,
<2> just joined StopStalk.
Send a friend request now - <3>
To stop receiving mails - <4>
```
where
<1> = row.stopstalk_handle
<2> = form.vars.first_name + " " + form.vars.last_name
<3> = URL('default', 'search')
<4> = URL("default", "unsubscribe", scheme=True, host=True, extension=False) [see here](https://github.com/stopstalk/stopstalk-deployment/blob/master/controllers/default.py#L906)

File: models/db.py [hunk 4]
```diff
@@ -220,6 +250,7 @@ def register_callback(form):
 
 auth.settings.register_onvalidation = [sanitize_fields]
 auth.settings.register_onaccept.append(register_callback)
+auth.settings.register_onaccept = [inside_institute]
```
Comment (raj454raj) [inline models/db.py]: Definitely wrong - 
```
auth.settings.register_onaccept.append(inside_institute)
```

File: models/db.py [hunk 5]
```diff
@@ -184,7 +184,37 @@ def sanitize_fields(form):
...
+                     form.vars.email,
+                     form.vars.institute,
+                     form.vars.stopstalk_handle,
+                     form.vars.referrer,
+                     form.vars.codechef_handle,
+                     form.vars.codeforces_handle,
+                     form.vars.spoj_handle,
+                     form.vars.hackerearth_handle,
+                     form.vars.hackerrank_handle)
+    send_mail(to=to, subject=subject, message=message)
```
Comment (raj454raj) [inline models/db.py]: send_mail function does not take a list of `to`'s.
So put every thing in the above loop (`for row in rows`)
and call send_mail in every iteration with a seperate email id.
O wait - move subject variable and message outside the loop and just change the `to`'s and then call send_mail in every iteration with a different `to` and same subject and message

**Comment by raj454raj** [comment]
@gauthamzz : Nice start !
Please do the modifications I suggested.
I would say please test the code you send a PR for - It is very necessary.
Thanks.

**Comment by gauthamzz** [comment]
for some unknown reason web2py is not working on my pc. I would have tested.im sorry for the inconvenience

**Comment by raj454raj** [comment]
There is no inconvenience. You are helping me and I am helping you ;) 
You can just clone the web2py repository and get going as mentioned in the README of this repository. Maybe able to help if you can show the error log or screenshots?

**Comment by gauthamzz** [comment]
now i fixed some of the errors. now it is working but it is saying unable to send email when a new user is registered

**Comment by raj454raj** [comment]
You will have to install postfix in whatever OS you are using.

**Comment by gauthamzz** [comment]
<type 'exceptions.AttributeError'> 'thread._local' object has no attribute 'mysql_user'
idk why this is coming.Do you have any idea ?

can i suggest you put use python2 in readme

**Comment by raj454raj** [comment]
O okay got it - 
copy the modules/0firstrun.py to modules/0firstrun.py and then open the file and add your mysql credentials there. I think you will need to create `migration` database in mysql also.
Which OS are you using?

**Comment by gauthamzz** [comment]
Linux mint . 
copy from the file to itself . I didnt get you

**Comment by raj454raj** [comment]
Sorry models* . Edited reply.

**Comment by gauthamzz** [comment]
New error came again . i think im wasting your time . i will send a PR with your suggested edits cos more errors are comming again and again .please test the code 
screenshot from 2016-04-20 04 03 27

**Comment by raj454raj** [comment]
Read the error man - Your mysql password is wrong(What you entered in 0firstrun.py). First try to open mysql alone in your terminal.

In your terminal - 
```
$ mysql -u root -p
```
Try entering your password.

**Comment by gauthamzz** [comment]
Well i did that already still the error comes

**Comment by raj454raj** [comment]
In your terminal - 
```
$ mysql -u root -p
```
Enter your password.
If you are in the mysql console.
Do - 
"""
mysql> USE migration;
mysql> SHOW TABLES;
"""
If here you can some tables then I guess it should work.
Also restart your web2py server whenever you are making changes in any files in `models/` or `modules`

Report any errors here.

**Comment by gauthamzz** [comment]
there is no database named migration

**Comment by raj454raj** [comment]
Great.
```
mysql> CREATE DATABASE migration;
```
And restart your web2py server.

**Pull request closed without merging**